numba/np/polynomial.py

```diff
@@ -57,3 +58,116 @@ def roots_impl(p):
             return roots
 
     return roots_impl
+
+
+@register_jitable
+def _trimseq(seq):
+    """Remove small Poly series coefficients.
+
+    Parameters
+    ----------
+    seq : sequence
+        Sequence of Poly series coefficients. This routine fails for
+        empty sequences.
+
+    Returns
+    -------
+    series : sequence
+        Subsequence with trailing zeros removed. If the resulting sequence
+        would be empty, return the first element. The returned sequence may
+        or may not be a view.
+
+    Notes
+    -----
+    Do not lose the type info if the sequence contains unknown objects.
+
+    """
+    if len(seq) == 0:
+        return seq
+    else:
+        for i in range(len(seq) - 1, -1, -1):
+            if seq[i] != 0:
+                break
+        return seq[:i+1]
+
+
+@overload(pu.trimseq)
+def polyutils_trimseq(seq):
+    if not isinstance(seq, types.Array):
+        msg = 'The argument "seq" must be an array'
+        raise errors.TypingError(msg)
+    
+    def impl(seq):
+        return _trimseq(seq)
+    
+    return impl
+
+
+@overload(poly.polyadd)
+def numpy_polyadd(c1, c2):
+    if not isinstance(c1, types.Array):
+        msg = 'The argument "c1" must be an array'
+        raise errors.TypingError(msg)
+    
+    if not isinstance(c2, types.Array):
+        msg = 'The argument "c2" must be an array'
+        raise errors.TypingError(msg)
```

Comment:
`c1` and `c2` must be array-like objects. See:

```python
>>> import numpy as np
>>> np.polynomial
>>> np.polynomial.polynomial.polyadd(1, 2)
array([3.])
```

In Numba there's a function called `type_can_asarray` that you can use to check if the type is convertible to an ndarray.